### PR TITLE
[Text] Add Drop Shadow Style to Text Component

### PR DESCRIPTION
**Pull request by greghe**
dropshadow

Adds four new styles to the Text component: `textShadowOpacity`,` textShadowColor`, `textShadowOffset` and `textShadowRadius`. They are nearly identical to the correspondingly named view shadow properties.

### CODE REVIEW

**Comment by ide** [comment]
LGTM. I'm wondering if we should drop the textShadowOpacity/shadowOpacity prop since the shadowColor should take an rgba value anyway. @vjeux do you agree with that or know why shadow opacity and color were separate props?

File: Libraries/Text/RCTShadowText.m [hunk 1]
```diff
@@ -201,6 +202,22 @@ - (NSAttributedString *)_attributedStringWithFontFamily:(NSString *)fontFamily
   if (_color) {
     [self _addAttribute:NSForegroundColorAttributeName withValue:_color toAttributedString:attributedString];
   }
+  
+  if (!isnan(_textShadowOpacity)) {
+      NSShadow *shadow = [[NSShadow alloc] init];
```
Comment (ide) [inline Libraries/Text/RCTShadowText.m]: (new coding guideline) `[NSShadow new]`

File: Libraries/Text/RCTShadowText.m [hunk 2]
```diff
@@ -180,7 +181,7 @@ - (NSAttributedString *)_attributedStringWithFontFamily:(NSString *)fontFamily
   if (!isnan(_letterSpacing)) {
     letterSpacing = @(_letterSpacing);
   }
-
+
```
Comment (ide) [inline Libraries/Text/RCTShadowText.m]: remove

**Comment by greghe** [comment]
I think there should be separate opacity and color styles for text shadows, just as there are separate `shadowColor` and `shadowOpacity` styles for views. For one, it makes it easy for the user (and the software) to toggle the shadows on and off completely.

**Comment by ide** [comment]
I'm proposing to consolidate the props for Views too. You could just as easily specify `shadowColor: 'transparent'` to quickly toggle the shadows.

**Comment by vjeux** [comment]
Would be nice to follow the CSS-spec naming convention in case of conflicts.

https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow
https://developer.mozilla.org/en-US/docs/Web/CSS/text-shadow

In this case, yeah, I agree that having an rgba color would be great

**Comment by ide** [comment]
@vjeux for the shadow properties, CSS only has the shorthand forms where you have to specify offset, radius, and color in one string together but I don't think we want that. I do think it would be good to support multiple shadows so maybe the syntax we want looks like this:

```js
textShadow: { offset, radius, color }
// and also
textShadow: [
  { offset1, radius1, color1 },
  { offset2, radius2, color2 }
]
```

**Comment by vjeux** [comment]
sounds good with me :)

**Comment by ide** [comment]
@greghe how does the proposal in my comment above sound? Initially we can support just one shadow instead of several ones, but I would like to start with the API we want so that we don't have to deprecate an old one or support several ways to do the same thing.

**Comment by greghe** [comment]
Sure, that sounds fine.

**Comment by jim-lake** [comment]
Some form of this would be very useful, text shadows are used heavily in native apps.

**Comment by ide** [comment]
@greghe can you update this PR to implement the `textShadow: { offset, radius, color }` API? Feel free to ping me if the API is unclear.

**Comment by mkonicek** [comment]
How is this going to work on Android? Can it have the same API?

**Comment by satya164** [comment]
@greghe Any updates on this?

**Comment by satya164** [comment]
Closing in favour of #4975 which also add Android support.

**Pull request closed without merging**